spec/delayed/priority_spec.rb

```diff
@@ -43,6 +45,17 @@
       expect(described_class.reporting).to eq 30
     end
 
+    context 'when assign_at_midpoint is set to true' do
+      let(:assign_at_midpoint) { true }
+
+      it 'returns the midpoint value' do
+        expect(described_class.interactive).to eq 5
+        expect(described_class.user_visible).to eq 15
+        expect(described_class.eventual).to eq 25
+        expect(described_class.reporting).to eq 35
+      end
```

Comment:
I think we also want to test the inverse of these.

There is a test down near line 110 that looks like this:
```ruby
  it 'provides the name of the priority range' do
    expect(described_class.new(0).name).to eq :interactive
    expect(described_class.new(3).name).to eq :interactive
    expect(described_class.new(10).name).to eq :user_visible
    expect(described_class.new(29).name).to eq :eventual
    expect(described_class.new(999).name).to eq :reporting
    expect(described_class.new(-123).name).to eq nil
  end
```

We should test with the midpoint feature enabled too, to ensure that the ranges still map to the same words. (E.g. 0-9 maps to interactive, 10-19 maps to user_visible, etc).

As it stands I suspect that 5-14 will map to interactive and 15-24 will map to user_visible (etc) and I want to either prove that or disprove that with the test coverage. 🙂

Reply:
Good call! this was not working.. the ranges were still being built using the new mid-point values. Fixed that by creating a new mapping altogether.